certbot-apache/certbot_apache/assertions.py

```diff
@@ -0,0 +1,116 @@
+        # We might want to disable this later if testing with two separate
+        # (but identical) directory structures.
+        assert first.filepath == second.filepath
+
+def assertEqualComment(first, second): # pragma: no cover
+    """ Equality assertion for CommentNode """
+
+    assert isinstance(first, interfaces.CommentNode)
+    assert isinstance(second, interfaces.CommentNode)
+
+    if not isPass(first.comment) and not isPass(second.comment):
+        assert first.comment == second.comment
+
+def _assertEqualDirectiveComponents(first, second): # pragma: no cover
+    """ Handles assertion for instance variables for DirectiveNode and BlockNode"""
+
+    # Enabled value cannot be asserted, because Augeas implementation
+    # is unable to figure that out.
+    # assert first.enabled == second.enabled
+    if not isPass(first.name) and not isPass(second.name):
+        assert first.name == second.name
+
+    if not isPass(first.parameters) and not isPass(second.parameters):
+        assert first.parameters == second.parameters
+
+def assertEqualDirective(first, second):
+    """ Equality assertion for DirectiveNode """
+
+    assert isinstance(first, interfaces.DirectiveNode)
+    assert isinstance(second, interfaces.DirectiveNode)
+    _assertEqualDirectiveComponents(first, second)
+
+def assertEqualBlock(first, second):
+    """ Equality assertion for BlockNode """
+
+    # first was checked in the assertEqual method
+    assert isinstance(first, interfaces.BlockNode)
+    assert isinstance(second, interfaces.BlockNode)
+    _assertEqualDirectiveComponents(first, second)
+    # Children cannot be asserted, because Augeas implementation will not
+    # prepopulate the sequence of children.
+    # assert len(first.children) == len(second.children)
+
+def isPass(value): # pragma: no cover
+    """Checks if the value is set to PASS"""
+    return PASS in value
+
+def isPassDirective(block):
+    """ Checks if BlockNode or DirectiveNode should pass the assertion """
+
+    if block.name == PASS:
+        return True
+    if PASS in block.parameters: # pragma: no cover
+        return True
+    if block.filepath == PASS: # pragma: no cover
+        return True
+    return False
+
+def isPassComment(comment):
+    """ Checks if CommentNode should pass the assertion """
+
+    if comment.comment == PASS:
+        return True
+    if comment.filepath == PASS: # pragma: no cover
+        return True
+    return False
+
+def isPassNodeList(nodelist): # pragma: no cover
+    """ Checks if a ParserNode in the nodelist should pass the assertion,
+    this function is used for results of find_* methods. Unimplemented find_*
+    methods should return a sequence containing a single ParserNode instance
+    with assertion pass string."""
+
+    try:
+        node = nodelist[0]
+    except IndexError:
+        node = None
+
+    if not node: # pragma: no cover
+        return False
+
+    if isinstance(node, interfaces.BlockNode):
```

Comment:
since a `BlockNode` is an instance of a `DirectiveNode`, this is unnecessary.

Reply:
Nice catch! This is a remnant from earlier state where there was different checks on `BlockNode`. It's now removed.